misc/python/materialize/mzbuild.py

```diff
@@ -48,6 +49,78 @@
 from materialize.xcompile import Arch, target
 
 
+class RustICE(Exception):
+    pass
+
+
+def run_and_detect_rust_ice(
+    cmd: list[str], cwd: str | Path
+) -> subprocess.CompletedProcess:
+    """This function is complex since it prints out each line immediately to
+    stdout/stderr, but still records them at the same time so that we can scan
+    for the Rust ICE."""
+    stdout_result = ""
+    stderr_result = ""
+    p = subprocess.Popen(
+        cmd,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        text=True,
+        bufsize=1,
+    )
+
+    sel = selectors.DefaultSelector()
+    sel.register(p.stdout, selectors.EVENT_READ)  # type: ignore
+    sel.register(p.stderr, selectors.EVENT_READ)  # type: ignore
+    assert p.stdout is not None
+    assert p.stderr is not None
+    os.set_blocking(p.stdout.fileno(), False)
+    os.set_blocking(p.stderr.fileno(), False)
+    while True:
```

Comment:
What is this `while True` loop for? Maybe I'm just missing something, but I don't see how we loop through it more than once.

Did you mean to have the `p.wait()` and the code below it outside this loop?

Reply:
Sorry, this commit was out of date, rebased.